Spellcheck

tommyod opened the pull request:
This PR is a continuation of my read-through of the docs, the earlier PRs are #18941, #18973 and #18948.

**The changes included in this PR include**
- Changed spelling of "python" to "Python" for consistency, same with "numpy" to "NumPy". This change was made globally, i.e. for most documents in `/docs/source/`.
- A review of `groupby.rst` and `missing_data.rst`
  - Added punctuation where missing.
  - Added more function references (i.e. `:meth:...`) where appropriate.
  - Rewrote a few sentences.

Feedback is very welcome.

tommyod committed:
Small changes in missing_data.rst

tommyod committed:
Started reviewing groupby.rst

tommyod committed:
Replaced the word 'python' with 'Python'

tommyod committed:
Replaced 'numpy' with 'NumPy'

tommyod committed:
Resolved merge conflicts

jreback commented:
> Changed spelling of "python" to "Python" for consistency, same with "numpy" to "NumPy". This change was made globally, i.e. for most documents in /docs/source/.

can you add a lint rule for this (and other things that we want to check globally)
in ``ci/lint.sh``

jreback commented:
also can you add a note in the whats new about documentation edits (there is a section) and include these PR#'s ?
and a similar note for #18966

tommyod committed:
Added documentation changes to whatsnew

tommyod commented:
@jreback I've added a reference to this and earlier PRs in `whatsnew/v0.23.0.txt`. I don't know how to make the linting rule, since for instance ` numpy ` should be replaced with ` NumPy ` in writte documentation, but not in the accompanying code. Not sure how to enforce this.

jreback commented:
rebase and push again, fixed some hanging by Travis CI

jreback commented on doc/source/cookbook.rst:
```diff
@@ -41,7 +41,7 @@ above what the in-line examples offer.
 explicitly imported for newer users.
 
-These examples are written for python 3.4.  Minor tweaks might be necessary for earlier python
+These examples are written for Python 3.4.  Minor tweaks might be necessary for earlier python
```
can you make this 3.6

jreback replied:
or rather say Python 3

jreback commented on doc/source/groupby.rst:
```diff
@@ -88,14 +88,14 @@ object (more on what the GroupBy object is later), you do the following:
-  - A dict or Series, providing a ``label -> group name`` mapping
+  - A Python function, to be called on each of the axis labels.
+  - A list or NumPy array of the same length as the selected axis.
+  - A dict or Series, providing a ``label -> group name`` mapping.
```
can you double-backticks Series/DataFrame when you see

jreback reviewed (changes requested):
small comments. can you add a rule in lint.sh?

tommyod committed:
Addressed review

tommyod commented:
@jreback - I've addressed the review, rebased and pushed. I could not figure out good logic for a rule in `lint.sh`, will come back to this in the future if I figure it out.

WillAyd commented:
@tommyod the grep command below could be a starter for you. This will find any instance of ``numpy`` but subsequently exclude it if the line contains ``from`` or ``import``

It could miss some lines like ``# need to import foo from numpy`` but should realistically catch most issues

`` grep -R --include="*.py" --include="*.pyx" --include="*.rst" "numpy" $path | grep -Ev "from|import"``

jreback commented:
thanks @tommyod 

yeah a rule similar to

``grep -R --include="*.py" --include="*.pyx" --include="*.rst" "numpy" doc/ pandas/ | grep -Ev "from|import|_numpy_|\/numpy\/"``

is close, though picking up extra things so prob have to add a bit to it

would take as a followup
